UWP build is broken

```
hkeyholder.hpp(37): error C2039: 'RegCloseKey': is not a member of '`global namespace'' (compiling source file telemetry_policy.cpp)
hkeyholder.hpp(37): error C3861: 'RegCloseKey': identifier not found (compiling source file telemetry_policy.cpp)
```

(It is HKeyHolder)

--- a/sdk/core/azure-core/inc/azure/core/internal/hkeyholder.hpp
+++ b/sdk/core/azure-core/inc/azure/core/internal/hkeyholder.hpp
@@ -8,6 +8,8 @@
  */
 #pragma once
 
+#include "azure/core/platform.hpp"
+
 #if defined(AZ_PLATFORM_WINDOWS)
 #if !defined(WIN32_LEAN_AND_MEAN)
 #define WIN32_LEAN_AND_MEAN
@@ -17,6 +19,9 @@
 #endif
 
 #include <windows.h>
+
+#if !defined(WINAPI_PARTITION_DESKTOP) \
+    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
 
 namespace Azure { namespace Core { namespace Internal {
 
@@ -54,5 +59,7 @@ namespace Azure { namespace Core { namespace Internal {
   };
 
 }}} // namespace Azure::Core::Internal
+
+#endif
 
 #endif

